modules/contracts/src.sol/transferDefinitions/ParameterizedPayments.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: UNLICENSED 
+pragma solidity ^0.7.1; 
+pragma experimental "ABIEncoderV2";
+
+import "../interfaces/ITransferDefinition.sol";
+import "../lib/LibChannelCrypto.sol";
+import "../lib/SafeMath.sol";
+import "../lib/strings.sol";
+
+/// @title ParameterizedPayments
+/// @notice This contract allows a series of payments to be bundled up
+///         as a single payment constrained to a set of specified parameters.
+///
+///         A parameterized payment can specify:
+///         - rate: how much value can be transferred over time
+///         - expiration: the latest time in which value can transfer
+///         - UUID: a unique identifier         
+
+contract ParameterizedPayments is ITransferDefinition {
+  using LibChannelCrypto for bytes32;
+  using SafeMath for uint256;
+  using strings for *;
+
+  // Rate = deltaAmount/deltaTime
+  struct Rate {
+    uint256 deltaAmount;
+    uint256 deltaTime;
+  }
+
+  // TransferState contains the payment data
+  struct TransferState {
+    address receiver;
+    uint256 start;
+    uint256 expiration;
+    bytes32 UUID;
+    Rate rate;
+  }
+
+  // Encapsulates the UUID and payment amount to make signing easier - see below TransferResolver
+  struct ResolverData {
+    bytes32 UUID;
+    uint256 paymentAmountTaken;
+  }
+
+  // TransferResolver contains the UUID of the payment we're signing to prevent replay attacks,
+  // the payment amount taken, and a signature of this data
+  struct TransferResolver {
+    ResolverData data;
+    bytes payeeSignature;
+  }
+
+  string RateEncoding = "tuple(uint256 deltaAmount, uint256 deltaTime)";
+  string StateEncoding = 
+    "tuple(address receiver, uint256 start, uint256 expiration, bytes32 UUID, "
+    .concat(RateEncoding)
+    .concat(")");
+  string ResolverDataEncoding = "tuple(bytes32 UUID, uint256 paymentAmountTaken)";
+  string ResolverEncoding = 
+    "tuple("
+    .concat(ResolverDataEncoding)
+    .concat(", bytes payeeSignature)");
```

Comment:
It would be a good idea to make sure these work (we've been burned before). It's not too bad to do it in the node repl, but would probably have to be done manually

Reply:
@LayneHaber can you give an overview of how I'd test these?

Reply:
Testing locally, will update.